library/alloc/src/vec/mod.rs

```diff
@@ -3656,20 +3656,22 @@ impl<T, A: Allocator> Vec<T, A> {
     ///
     /// ```
     /// # use std::cmp::min;
-    /// # let some_predicate = |x: &mut i32| { *x == 2 || *x == 3 || *x == 6 };
-    /// # let mut vec = vec![1, 2, 3, 4, 5, 6];
-    /// # let range = 1..4;
+    /// # let some_predicate = |x: &mut i32| { *x % 2 == 1 };
+    /// # let mut vec = vec![0, 1, 2, 3, 4, 5, 6];
+    /// # let range = 1..5;
     /// let mut i = range.start;
-    /// while i < min(vec.len(), range.end) {
+    /// let mut end = range.end;
+    /// while i < min(vec.len(), end) {
     ///     if some_predicate(&mut vec[i]) {
     ///         let val = vec.remove(i);
+    ///         end -= 1;
```

Comment:
Isn't this a redundant op? `end` and `vec.len()` only ever change at the same time, so it should be possible to do something like `while i < vec.len() - diff`, where `diff` is `vec.len() - range.end` using the original `vec.len()`.